docs: complete Block Access, Subscribe, and Backfill flow stubs in data-flow.md (#3063)
hiero-ledger/hiero-block-node: docs/block-node/architecture/data-flow.md

   `block_number` or sets `retrieve_latest` to `true`.
2. The plugin resolves the target block number: an explicit `block_number ≥ 0` is used directly; `retrieve_latest = true`
   (or `block_number = -1`) resolves to the maximum block number returned by `HistoricalBlockFacility.availableBlocks().max()`.
   An unrecognised combination returns `INVALID_REQUEST`.
3. The plugin calls `availableBlocks().contains(N)` to confirm the resolved block number is within the current stored
   range. If not, it returns `NOT_AVAILABLE` without touching storage.
4. The plugin calls `blockProvider.block(N)`, which opens a `BlockAccessor` pointing at the requested block.
   `HistoricalBlockFacility` dispatches internally to whichever storage tier currently holds that block:
   `BlockFileRecentPlugin` (backed by fast NVMe, holds recent and live-stream blocks) for newer blocks, or
   `BlockFileHistoricPlugin` (backed by bulk HDD, holds the compressed historic archive) for deep history.
   Blocks in either tier were verified by `VerificationServicePlugin` before persistence, so any successful

>    response contains a fully verified block including its `BlockProof`.

Reviewer: 1. The two _current_ block storage plugins are just what we've built so far; anyone (us or others) can, and hopefully will, build other options.
1. The prior verification is irrelevant and _should not_ be considered.  The Proofs are _always_ present (one or more per block); and recipients should never trust the source as "verified".

    participant Facility as HistoricalBlockFacility
    participant Messaging as BlockMessagingFacility

    Sub->>Plugin: subscribeBlockStream(SubscribeStreamRequest)
    Plugin->>Handler: apply(request, pipeline)
    Handler->>Session: create(clientId, request) and submit to virtual thread
    Note over Handler: await sessionReadyLatch
    Session->>Session: validate request and determine mode
    Session-->>Handler: sessionReadyLatch released

    opt Historical phase (modes: first-available-to-end, from-start, bounded-range)
        loop Blocks N through latest_persisted
            Session->>Facility: block(N)
            Facility-->>Session: BlockUnparsed
            Session-->>Sub: block_items + end_of_block
        end
    end

    opt Live phase (modes: live-only, from-start-indefinitely)
        Session->>Messaging: register LiveBlockHandler
        loop Each new live block published
            Messaging-->>Session: block items via LMAX Disruptor
            Session-->>Sub: block_items + end_of_block
        end
    end

    Session-->>Sub: status(SUCCESS)
    Note over Sub: On disconnect, client resubscribes<br/>from last received block + 1
```

Detailed Steps:

> 1. A subscriber (typically a Mirror Node) sends a `subscribeBlockStream` server-streaming request specifying

Reviewer: We probably do not want to specify a "typical" subscriber, we don't know what that will look like when the network adoption grows.

```suggestion
1. A subscriber sends a `subscribeBlockStream` server-streaming request specifying
```

     from the newest block published.
   - **First-available-to-end** — `start = uint64_max`, explicit `end = M`: serve from the earliest stored block
     through block M.
   - **From-start-indefinitely** — explicit `start = N`, `end = uint64_max`: serve history from block N, then
     transition to live.
   - **Bounded-range** — explicit `start = N`, explicit `end = M` (M ≥ N): serve blocks N through M and close.
4. **Historical phase** (modes 2, 3, and 4): the session reads blocks sequentially from `HistoricalBlockFacility`.
   Each block is serialized as one or more `BlockItemSet` messages (`block_items`) followed by a single `end_of_block`
   response carrying the block number.
5. **Live phase** (modes 1 and 3): the session registers a `LiveBlockHandler` (implementing
   `NoBackPressureBlockItemHandler`) on `BlockMessagingFacility`. New blocks published by `StreamPublisherPlugin`
   travel through `BlockMessagingFacility`'s LMAX Disruptor ring buffer and are forwarded to the subscriber
   pipeline in real time. The handler is "no back-pressure" because slow subscribers are disconnected rather

>    than blocking the publish path.

Reviewer: This is not correct.

```suggestion
   pipeline in real time. The handler is "no back-pressure" because slow subscribers are switched between
   "historical" and "live" as needed rather than blocking the publish path.
```

    Runner->>Fetcher: selectNextChunk(currentBlock, gapEnd, availability)
    Fetcher->>Fetcher: PriorityHealthBasedStrategy.select()
    Fetcher-->>Runner: NodeSelection (peer and chunk range)
    Runner->>Fetcher: fetchBlocksFromNode(peer, chunk)
    Fetcher->>Peer: subscribeBlockStream(start=N, end=M)

    alt Fetch succeeds
        Peer-->>Fetcher: block stream
        Fetcher->>Fetcher: markSuccess — reset failures, record latency in healthMap
        Fetcher-->>Runner: List of BlockUnparsed
        loop Each fetched block
            Runner->>Messaging: sendBackfilledBlockNotification
        end
        Messaging-->>Plugin: PersistedNotification (backfill source)
        Plugin->>Plugin: decrement pending, increment backfilled counters
    else Fetch fails (error or timeout)
        Fetcher->>Fetcher: markFailure — increment failures, apply backoff in healthMap
        Note over Runner: retry with next best peer (up to maxRetries)
    end
```

Detailed Steps:

1. `BackfillPlugin` runs two complementary gap-detection triggers. A periodic autonomous scanner runs
   `detectAndScheduleGaps()` on a `scheduleAtFixedRate` loop (interval: `backfillConfiguration.scanInterval()`).

>    In addition, when `BlockMessagingFacility` broadcasts a `NewestBlockKnownToNetworkNotification` (a new block seen

Reviewer: It's Publisher that sends the notification.

```suggestion
   In addition, when the `stream-publisher` plugin broadcasts a `NewestBlockKnownToNetworkNotification` (a newer block seen
```
